src/inputs.rs

```diff
@@ -354,43 +445,39 @@ impl<'source> FromPyObject<'source> for StatefulSource {
 }
 
 impl StatefulSource {
-    fn next(&self, py: Python) -> PyResult<Option<Vec<TdPyAny>>> {
-        match self.0.call_method0(py, "next") {
+    fn next_batch(&self, py: Python) -> PyResult<Option<Vec<TdPyAny>>> {
+        match self.0.call_method0(py, intern!(py, "next_batch")) {
             Err(stop_ex) if stop_ex.is_instance_of::<PyStopIteration>(py) => Ok(None),
             Err(err) => Err(err),
-            Ok(items) => Ok(Some(
-                items
-                    .extract(py)
-                    .reraise("`StatefulSource.next` did not return a list")?,
-            )),
+            Ok(items) => Ok(Some(items.extract(py).reraise(
+                "`next_batch` method of StatefulSource did not return a list",
+            )?)),
         }
     }
 
     fn next_awake(&self, py: Python) -> PyResult<Option<DateTime<Utc>>> {
-        self.0
-            .call_method0(py, "next_awake")
-            .reraise("error calling `StatelessSource.next_awake`")?
-            .extract(py)
-            .reraise("error converting `StatefulSource.next_awake` return value to UTC datetime")
+        Ok(self
+            .0
+            .call_method0(py, intern!(py, "next_awake"))?
+            .extract(py)?)
```

Comment:
```rust
        self.0
            .call_method0(py, intern!(py, "next_awake"))?
            .extract(py)
```

Reply:
There are a number of fixes like this that you can spot with `clippy`, see main PR comment